modules/preloader/injected/EthereumProvider.js

```diff
@@ -1,191 +1,201 @@
+const origin = this.origin;
+
 (function() {
-  var EventEmitter = window.EventEmitter;
+  class EthereumProvider extends window.EventEmitter {
+    constructor() {
+      // Call super for `this` to be defined
+      super();
+
+      // Init storage
+      this._isConnected = false;
+      this._nextJsonrpcId = 0;
+      this._promises = {};
+      this._activeSubscriptions = [];
+
+      // Fire the connect
+      this._connect();
+
+      // Listen for jsonrpc responses
+      window.addEventListener('message', this._handleJsonrpcMessage.bind(this));
+    }
+
+    /* Methods */
 
-  var postMessage = function(payload) {
-    if (typeof payload === 'object') {
-      payload = JSON.stringify(payload);
+    enable() {
+      return new Promise((resolve, reject) => {
+        window.mist
+          .requestAccounts()
+          .then(resolve)
+          .catch(reject);
+      });
     }
 
-    window.postMessage(
-      payload,
-      !location.origin || location.origin === 'null' ? '*' : location.origin
-    );
-  };
+    send(method, params = []) {
+      if (!method || typeof method !== 'string') {
+        return new Error('Method is not a valid string.');
+      }
+
+      if (!(params instanceof Array)) {
+        return new Error('Params is not a valid array.');
+      }
+
+      const id = this._nextJsonrpcId++;
+      const jsonrpc = '2.0';
```

Comment:
const id and jsonrpc is just used once for the const payload and because of that I would declare it like this:
```const payload { '2.0', this._nextJsonrpcId++, method, params }```